tests/unit/test_charm.py

```diff
@@ -81,29 +68,27 @@ def read(self, *args, **kwargs):
 ]
 
 
+@patch.object(Container, "restart", new=lambda x, y: True)
 class TestCharm(unittest.TestCase):
     @patch("charm.KubernetesServicePatch", lambda x, y: None)
     def setUp(self):
         self.harness = Harness(GrafanaAgentOperatorCharm)
         self.addCleanup(self.harness.cleanup)
         self.harness.set_model_info(name="lma", uuid="1234567890")
         self.harness.set_leader(True)
-        self.harness.begin()
+        self.harness.begin_with_initial_hooks()
```

Comment:
[Apparently](https://github.com/canonical/operator/pull/727#issuecomment-1079096588) we need to call `pebble_ready` ourselves after `begin_with_initial_hooks`.
Since there is an `except (FileNotFoundError, PathError)` above, it would be nice to test the outcome before and after calling `pebble_ready`.

Reply:
Resolving because itests are passing

Reply:
Do you mind rerunning the tests with the following added after `begin_with_initial_hooks`?
`self.harness.container_pebble_ready("agent")`

Reply:
The tests pass either way. Should I add the line permanently?